developer-extension/src/panel/components/actionsBar.tsx

```diff
@@ -1,45 +1,56 @@
 import { Group, Checkbox, Badge, Button } from '@mantine/core'
-import React from 'react'
+import { useInterval } from 'usehooks-ts'
+import React, { useEffect, useState } from 'react'
 import { evalInWindow } from '../evalInWindow'
 import { useStore } from '../hooks/useStore'
 
 export function ActionsBar() {
   const [{ useDevBundles, useRumSlim, devServerStatus, blockIntakeRequests }, setStore] = useStore()
+  const [autoFlush, setAutoFlush] = useState<boolean>(true)
+
+  useEffect(flushEvents, [])
+  useInterval(flushEvents, 5000)
```

Comment:
<!-- Warning represent user-facing problems. If possible, it's great to follow this kind of comment with a suggestion.  -->
**🔨 warning:** ‏disabling auto flush won't work, as we don't check for the `autoFlush` variable

<!-- Warning represent user-facing problems. If possible, it's great to follow this kind of comment with a suggestion.  -->
**🔨 warning:** ‏usehooks-ts is not compatible with our nodejs version (kind of weird but well...)

<!-- Suggestions are specific requests to improve the subject under review. It is assumed that we all want to do what's best, so these comments are never dismissed as “mere suggestions”, but are taken seriously.  -->
**💬 suggestion:** ‏write a more well-suited hook ourselves:

```ts
function useAutoFlushEvents(enabled: boolean) {
  useEffect(() => {
    if (enabled) {
      flushEvents()
      const id = setInterval(flushEvents, FLUSH_EVENTS_INTERVAL)
      return () => clearInterval(id)
    }
  }, [enabled])
}


// then use it in the `ActionBar` component like that:
useAutoFlushEvents(autoFlush)
```